File: frontend/src/types/venue-map.ts
export type PlaceKind = 'seat' | 'section';

export interface VenueMapPlace {
  id: string;
  kind: PlaceKind;
  label: string;
  x: number;
  y: number;
  width?: number | null;
  height?: number | null;
  capacity: number;
  available: number;
  ticketTypeId: string;
}

export interface VenueMapDecoration {
  id: string;
  x: number;
  y: number;
  width: number;
  height: number;
  label: string;
}

export interface VenueMap {
  id: string;
  name: string;
  width: number;
  height: number;
  decorations: VenueMapDecoration[];
  places: VenueMapPlace[];
}

File: backend/Ticketing.Domain/Entities/VenueMap.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Ticketing.Domain.Entities;

public class VenueMap
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public required string Name { get; set; }
    public required int Width { get; set; }
    public required int Height { get; set; }
    public Guid CreatedBy { get; set; }
    public User CreatedByUser { get; set; } = null!;

    [Timestamp]
    public byte[] RowVersion { get; set; } = [];

    public ICollection<VenueMapPlace> Places { get; set; } = [];
    public ICollection<VenueMapDecoration> Decorations { get; set; } = [];
}
